### PR TITLE
dpdkdrv: improve unit test coverage

### PR DESCRIPTION
**Pull request by mythi**
Add NewDevicePlugin() tests to improve test coverage. This also
contributes to "input validation" (part of #321) that wasn't done
properly before.

Fixes: #325

Signed-off-by: Mikko Ylinen <mikko.ylinen@intel.com>

### CODE REVIEW

File: cmd/qat_plugin/dpdkdrv/dpdkdrv_test.go [hunk 1]
```diff
@@ -63,6 +63,49 @@ func createTestFiles(prefix string, dirs []string, files map[string][]byte, syml
...
+			expectedErr:     false,
+		},
+	}
+	for _, tt := range tcases {
+
+		_, err := NewDevicePlugin(1, tt.kernelVfDrivers, tt.dpdkDriver)
+
+		if tt.expectedErr && err == nil {
+			t.Errorf("Test case '%s': expected error, but got success", tt.name)
+		}
```
Comment (rojkov) [inline cmd/qat_plugin/dpdkdrv/dpdkdrv_test.go]: Also we should check if `err != nil` and we expect no error.

Comment (mythi) [reply cmd/qat_plugin/dpdkdrv/dpdkdrv_test.go]: yes, of course

**Comment by mythi** [comment]
FWIW, the coverage improves from ~74% to 87%

File: cmd/qat_plugin/dpdkdrv/dpdkdrv_test.go (hunk 1 above)
Comment (mythi) [reply cmd/qat_plugin/dpdkdrv/dpdkdrv_test.go]: fixed

**Comment by mythi** [comment]
v2 moves to use `t.Run()`.

**Pull request merged**